pgmpy/factors/CPD.py

```diff
@@ -110,9 +110,12 @@ def __init__(self, variable, variable_card, values,
                                     " of strings.")
             self.evidence = evidence
             variables.extend(evidence[::-1])
-            if not len(evidence_card) == len(evidence):
+            if len(evidence_card) < len(evidence):
```

Comment:
@yashu-seth We can change the error message to a more clear one. But adding an extra condition doesn't make much sense to me. Since, it would be very few times that these errors would be thrown and for the normal case we will be adding an extra condition which will be run every time a CPD is defined.